OsmAnd/src/net/osmand/plus/plugins/monitoring/MonitoringSettingsFragment.java

```diff
@@ -531,6 +539,11 @@ public void onDisplayPreferenceDialog(Preference preference) {
 				ApplicationMode appMode = getSelectedAppMode();
 				SingleSelectPreferenceBottomSheet.showInstance(fm, prefId, this, false, appMode, true, true);
 			}
+		} else if (prefId.equals(RECORD_OBD_DATA)) {
```

Comment:
CONSTANT.equals() or Algorythms.strings/object equals() (in case prefId is null)